project_template/tests/test_project_template.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2019 Patrick Wilson <patrickraymondwilson@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests import common
+
+
+class TestProjectTemplate(common.TransactionCase):
+
+    # Use case : Prepare some data for current test case
+    def setUp(self):
+        super().setUp()
+        test_project = self.env['project.project'].create({
+            'name': 'TestProject'})
+        self.env['project.task'].create({
+            'name': 'TestTask',
+            'project_id': test_project.id})
+
+    # TEST 01: Set project to be a template
+    def test_template(self):
+        test_project.is_template = True

```

Comment:
test_project cannot to be reached from here it's out of scope. You need to make an instance variable using ```self```.  In ```setUp``` do self.test_project instead.

To run unit tests locally add --test-enable -u project_template -d {mydb}. Replace {mydb} with your database name. You can find more info here: https://www.odoo.com/documentation/11.0/reference/testing.html